op-challenger/challenger/log_store_test.go

```diff
@@ -0,0 +1,225 @@
+}
+
+// TestLogStore_Subscribe_MissingClient tests the [Subscribe] method on a [logStore]
+// fails when the client is missing.
+func TestLogStore_Subscribe_MissingClient(t *testing.T) {
+	query := ethereum.FilterQuery{}
+	log := testlog.Logger(t, log.LvlError)
+	logStore := NewLogStore(query, nil, log)
+	err := logStore.Subscribe()
+	require.EqualError(t, err, ErrMissingClient.Error())
+}
+
+// TestLogStore_Quit tests the [Quit] method on a [logStore].
+func TestLogStore_Quit(t *testing.T) {
+	query := ethereum.FilterQuery{}
+	client := newMockLogStoreClient()
+	log := testlog.Logger(t, log.LvlError)
+	logStore := NewLogStore(query, client, log)
+
+	// A nil subscription should not cause a panic.
+	logStore.subscription = nil
+	logStore.Quit()
+
+	// Subscribe to the logStore.
+	err := logStore.Subscribe()
+	require.NoError(t, err)
+
+	// Quit the subscription
+	logStore.Quit()
+	require.Nil(t, logStore.subscription)
+}
+
+// TestLogStore_Resubsribe tests the [Resubscribe] method on a [logStore].
+func TestLogStore_Resubscribe(t *testing.T) {
+	query := ethereum.FilterQuery{}
+	client := newMockLogStoreClient()
+	log := testlog.Logger(t, log.LvlError)
+	logStore := NewLogStore(query, client, log)
+
+	// Subscribe to the logStore.
+	err := logStore.Subscribe()
+	require.NoError(t, err)
+
+	// Resubscribe to the logStore.
+	err = logStore.resubscribe()
+	require.NoError(t, err)
+}
+
+// TestLogStore_Logs tests log methods on a [logStore].
+func TestLogStore_Logs(t *testing.T) {
+	query := ethereum.FilterQuery{}
+	client := newMockLogStoreClient()
+	log := testlog.Logger(t, log.LvlError)
+	logStore := NewLogStore(query, client, log)
+
+	require.Equal(t, []types.Log{}, logStore.GetLogs())
+	require.Equal(t, []types.Log(nil), logStore.GetLogByBlockHash(common.HexToHash("0x1")))
+
+	// Insert logs.
+	logStore.insertLog(types.Log{
+		BlockHash: common.HexToHash("0x1"),
+	})
+	logStore.insertLog(types.Log{
+		BlockHash: common.HexToHash("0x1"),
+	})
+
+	// Validate log insertion.
+	require.Equal(t, 2, len(logStore.GetLogs()))
+	require.Equal(t, 2, len(logStore.GetLogByBlockHash(common.HexToHash("0x1"))))
+}
+
+// TestLogStore_DispatchLogs tests the [DispatchLogs] method on the [logStore].
+func TestLogStore_DispatchLogs(t *testing.T) {
+	query := ethereum.FilterQuery{}
+	client := newMockLogStoreClient()
+	log := testlog.Logger(t, log.LvlError)
+	logStore := NewLogStore(query, client, log)
+
+	// Subscribe to the logStore.
+	err := logStore.Subscribe()
+	require.NoError(t, err)
+
+	// Dispatch logs on the logStore.
+	go logStore.dispatchLogs()
+
+	// Send logs through the subscription.
+	blockHash := common.HexToHash("0x1")
+	logStore.subscription.logs <- types.Log{
+		BlockHash: blockHash,
+	}
+
+	// Wait for the logs to be dispatched.
+	timeout, tCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer tCancel()
+	err = e2eutils.WaitFor(timeout, 500*time.Millisecond, func() (bool, error) {
```

Comment:
Very low timeout.

Reply:
I think that 500ms is how often to run the verification and the timeout is actually the 30s on the context.